spaceMissionManager-service/src/main/java/cz/muni/fi/services/impl/UserServiceImpl.java

```diff
@@ -23,21 +23,39 @@
     UserDao userDao;
 
     @Override
-    public void acceptAssignedMission(Long userid){
-        User user = userDao.findUserById(userid);
+    public void acceptAssignedMission(User user){
         if(user.missionStatusPending()){
             user.setAcceptedMission(true);
+        } else {
+            throw new IllegalArgumentException("This user does not have pending mission status.");
+        }
+        try {
+            userDao.updateUser(user);
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not update user.", e);
         }
     }
 
     @Override
-    public void rejectAssignedMission(Long userid, String explanation) throws IllegalArgumentException{
+    public void rejectAssignedMission(User user, String explanation) throws IllegalArgumentException{
```

Comment:
You don't want to inform anyone that you are throwing IllegalArgumentException, because it is a RuntimeException